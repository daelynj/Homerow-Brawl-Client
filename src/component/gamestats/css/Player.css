.Player {
  float: left;
  padding-top: 10px;
  padding-left: 10px;
}
