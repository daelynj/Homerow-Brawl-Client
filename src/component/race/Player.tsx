import * as React from "react";
import octane from "./images/octane.png";

interface Props {
  position: number;
  name: string;
}

export const Player = (props: Props) => (
  <div>
    {props.name}
    <img
      style={{
        paddingLeft: props.position + "%"
      }}
      src={octane}
      alt="Octane"
    />
  </div>
);
