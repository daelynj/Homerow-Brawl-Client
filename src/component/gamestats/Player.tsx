import * as React from "react";
import "./css/Player.css";

interface Props {
  name: string;
  WPM: number;
  accuracy: number;
  wordsTyped: number;
  time: number;
  mistakes: number;
  place: number;
}

export const Player = (props: Props) => {
  const rank = () => {
    let place = props.place + 1;
    var j = place % 10;
    var k = place % 100;

    if (j === 1 && k !== 11) {
      return place + "st";
    }
    if (j === 2 && k !== 12) {
      return place + "nd";
    }
    if (j === 3 && k !== 13) {
      return place + "rd";
    }
    return place + "th";
  };

  return (
    <span className="player">
      <div>{rank()}</div>
      <div className="player__name">{"name: " + props.name}</div>
      <div>{"WPM: " + props.WPM}</div>
      <div>{"accuracy: " + props.accuracy + "%"}</div>
      <div>{"words typed: " + props.wordsTyped}</div>
      <div>{"time: " + props.time + " seconds"}</div>
      <div>{"mistakes: " + props.mistakes}</div>
    </span>
  );
};
