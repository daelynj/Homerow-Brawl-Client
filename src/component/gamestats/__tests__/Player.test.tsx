import * as React from "react";
import { Player } from "../Player";
import renderer from "react-test-renderer";

describe("Player", () => {
  it("renders the described Player", () => {
    const tree = renderer
      .create(
        <Player
          name={"octane"}
          WPM={100}
          accuracy={75.0}
          wordsTyped={50}
          time={50}
          mistakes={3}
          place={0}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
