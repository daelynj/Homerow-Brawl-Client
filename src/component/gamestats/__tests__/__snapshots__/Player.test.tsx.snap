// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders the described Player 1`] = `
<span
  className="player"
>
  <div>
    1st
  </div>
  <div
    className="player__name"
  >
    name: octane
  </div>
  <div>
    WPM: 100
  </div>
  <div>
    accuracy: 75%
  </div>
  <div>
    words typed: 50
  </div>
  <div>
    time: 50 seconds
  </div>
  <div>
    mistakes: 3
  </div>
</span>
`;
